lib/formats/bs_format_ripejson.c

```diff
@@ -0,0 +1,679 @@
+/* ======================================================== */
+
+#define FIELDPTR(field) STATE->json_fields.field.ptr
+#define FIELDLEN(field) STATE->json_fields.field.len
+
+#define STRTOUL(field, dest)                        \
+  do {                                              \
+    char tmp=FIELDPTR(field)[FIELDLEN(field)];      \
+    FIELDPTR(field)[FIELDLEN(field)]='\0';          \
+    dest=strtoul((char*)FIELDPTR(field), NULL, 10); \
+    FIELDPTR(field)[FIELDLEN(field)]=tmp;           \
+  } while(0)
+#define STRTOD(field, dest)                         \
+  do {                                              \
+    char tmp=FIELDPTR(field)[FIELDLEN(field)];      \
+    FIELDPTR(field)[FIELDLEN(field)]='\0';          \
+    dest=strtod((char*)FIELDPTR(field), NULL    );  \
+    FIELDPTR(field)[FIELDLEN(field)]=tmp;           \
+  } while(0)
+
+// convert char array to bytes
+// by @alistair
+static int hexstr_to_bytes(uint8_t *buf, const char* hexstr, size_t hexstr_len){
+  size_t i;
+  char c;
+  for (i = 0; i < hexstr_len; i++) {
+    c = hexstr[i];
+    // sanity check on input characters
+    if (c < '0' || (c > '9' && c < 'A') || (c > 'F' && c < 'a') || c > 'f') {
+      return -1;
+    }
+    if (c >= 'a') {
+      c -= ('a' - '9' - 1);
+    } else if (c >= 'A') {
+      c -= ('A' - '9' - 1);
+    }
+    c -= '0';
+
+    if ((i & 0x1) == 0) {
+      // high-order
+      *buf = c << 4;
+    } else {
+      // low-order
+      *(buf++) |= c;
+    }
+  }
+  return 0;
+}
+
+// convert bgp message hex string to char (byte) array, with added header marker
+// by @alistairking
+static ssize_t hexstr_to_bgpmsg(uint8_t *buf, size_t buflen,
+                                const char* hexstr, size_t hexstr_len,
+                                uint8_t msg_type)
+{
+  // 2 characters per octet, and BGP messages cannot be more than 4096 bytes
+  if ((hexstr_len & 0x1) != 0 || hexstr_len > (4096*2)){
+    return -1;
+  }
+  uint16_t msg_len = (hexstr_len / 2) + 2 + 1;
+  uint16_t tmp = htons(msg_len);
+  uint8_t *bufp = buf;
+
+  if (msg_len > buflen) {
+    return -1;
+  }
+
+  // populate the message header (but don't include the marker)
+  memcpy(buf, &tmp, sizeof(tmp));
+  buf[2] = msg_type;
+
+  // parse the hex string, one nybble at a time
+  bufp = buf + 3;
+  if(hexstr_to_bytes(bufp, hexstr, hexstr_len)<0){
+    return -1;
+  }
+
+  return msg_len;
+}
+
+/* ====================================================================== */
+/* ====================================================================== */
+/* ==================== PRIVATE FUNCTIONS BELOW HERE ==================== */
+/* ====================================================================== */
+/* ====================================================================== */
+
+// process common header fields for all message types
+int process_common_fields(bgpstream_format_t *format, bgpstream_record_t *record){
```

Comment:
all these internal functions should be made static

Reply:
return signatures updated to `static bgpstream_format_status_t` https://github.com/CAIDA/libbgpstream/commit/2e06de09f22f52937375477d1c7a637775eb7732